diff --git a/qemu-kvm.spec b/qemu-kvm.spec
index c1f98a0..7ae341b 100644
--- a/qemu-kvm.spec
+++ b/qemu-kvm.spec
@@ -12,6 +12,7 @@
 %endif
 
 %global have_usbredir 1
+%global have_spice    1
 %global have_opengl   1
 %global have_fdt      1
 %global have_modules_load 0
@@ -82,6 +83,7 @@
 %ifarch x86_64
     %global kvm_target    x86_64
 %else
+    %global have_spice   0
     %global have_opengl  0
 %endif
 %ifarch %{power64}
@@ -108,6 +110,9 @@
 #Versions of various parts:
 
 %global requires_all_modules                                     \
+%if %{have_spice}                                                \
+Requires: %{name}-ui-spice = %{epoch}:%{version}-%{release}      \
+%endif                                                           \
 %if %{have_opengl}                                               \
 Requires: %{name}-ui-opengl = %{epoch}:%{version}-%{release}     \
 Requires: %{name}-ui-egl-headless = %{epoch}:%{version}-%{release}     \
@@ -129,8 +134,6 @@ Requires: %{name}-block-blkio = %{epoch}:%{version}-%{release}   \
 Requires: %{name}-block-rbd = %{epoch}:%{version}-%{release}     \
 Requires: %{name}-audio-pa = %{epoch}:%{version}-%{release}
 
-# Since SPICE is removed from RHEL-9, the following Obsoletes:
-# removes {name}-ui-spice for upgrades from RHEL-8
 # The "<= {version}" assumes RHEL-9 version >= RHEL-8 version (in
 # other words RHEL-9 rebases are done together/before RHEL-8 ones)
 
@@ -140,7 +143,6 @@ Requires: %{name}-audio-pa = %{epoch}:%{version}-%{release}
 # Note: ssh driver wasn't removed yet just disabled due to late handling
 
 %global obsoletes_some_modules                                  \
-Obsoletes: %{name}-ui-spice <= %{epoch}:%{version}                       \
 Obsoletes: %{name}-block-gluster <= %{epoch}:%{version}                  \
 Obsoletes: %{name}-block-iscsi <= %{epoch}:%{version}                    \
 Obsoletes: %{name}-block-ssh <= %{epoch}:%{version}                    \
@@ -528,7 +530,13 @@ BuildRequires: pkgconfig(gbm)
 BuildRequires: perl-Test-Harness
 BuildRequires: libslirp-devel
 BuildRequires: pulseaudio-libs-devel
+%if %{have_spice}
 BuildRequires: spice-protocol
+BuildRequires: spice-server-devel
+BuildRequires: libcacard-devel
+# For smartcard NSS support
+BuildRequires: nss-devel
+%endif
 BuildRequires: capstone-devel
 
 # Requires for qemu-kvm package
@@ -768,6 +776,18 @@ Obsoletes: %{name}-hw-usbredir <= %{epoch}:%{version}
 This package provides usbredir support.
 %endif
 
+%if %{have_spice}
+%package  ui-spice
+Summary: QEMU spice support
+Requires: %{name}-common%{?_isa} = %{epoch}:%{version}-%{release}
+%if %{have_opengl}
+Requires: %{name}-ui-opengl%{?_isa} = %{epoch}:%{version}-%{release}
+%endif
+
+%description ui-spice
+This package provides spice support.
+%endif
+
 %package  ui-dbus
 Summary: QEMU D-Bus UI driver
 Requires: %{name}-common%{?_isa} = %{epoch}:%{version}-%{release}
@@ -1026,6 +1046,11 @@ run_configure \
   --enable-tcg \
   --enable-tools \
   --enable-tpm \
+%if %{have_spice}
+  --enable-smartcard \
+  --enable-spice \
+  --enable-spice-protocol \
+%endif
 %if %{have_usbredir}
   --enable-usb-redir \
 %endif
@@ -1242,6 +1267,8 @@ rm -rf %{buildroot}%{_datadir}/%{name}/opensbi-riscv64-generic-fw_dynamic.*
 rm -rf %{buildroot}%{_datadir}/%{name}/qemu-nsis.bmp
 rm -rf %{buildroot}%{_datadir}/%{name}/npcm7xx_bootrom.bin
 
+rm -rf ${RPM_BUILD_ROOT}%{_libdir}/qemu-kvm/ui-spice-app.so
+
 # Remove virtfs-proxy-helper files
 rm -rf %{buildroot}%{_libexecdir}/virtfs-proxy-helper
 rm -rf %{buildroot}%{_mandir}/man1/virtfs-proxy-helper*
@@ -1512,6 +1539,17 @@ useradd -r -u 107 -g qemu -G kvm -d / -s /sbin/nologin \
     %{_libdir}/%{name}/hw-usb-redirect.so
 %endif
 
+%if 0%{have_spice}
+%files ui-spice
+    %{_libdir}/qemu-kvm/hw-usb-smartcard.so
+    %{_libdir}/qemu-kvm/audio-spice.so
+    %{_libdir}/qemu-kvm/ui-spice-core.so
+    %{_libdir}/qemu-kvm/chardev-spice.so
+%ifarch x86_64
+    %{_libdir}/qemu-kvm/hw-display-qxl.so
+%endif
+%endif
+
 %files audio-dbus
 %{_libdir}/%{name}/audio-dbus.so
 
