diff --git a/seabios.spec b/seabios.spec
index 4b10da0..6441cbb 100644
--- a/seabios.spec
+++ b/seabios.spec
@@ -10,6 +10,7 @@ Source0:        https://code.coreboot.org/p/seabios/downloads/get/seabios-1.16.3
 
 
 Source10:       config.vga-cirrus
+Source12:       config.vga-qxl
 Source13:       config.vga-stdvga
 Source18:       config.seabios-256k
 Source19:       config.vga-virtio
@@ -108,7 +109,7 @@ build_bios %{_sourcedir}/config.seabios-256k bios.bin bios-256k.bin
 
 
 # seavgabios
-%global vgaconfigs cirrus stdvga virtio ramfb bochs-display
+%global vgaconfigs cirrus qxl stdvga virtio ramfb bochs-display
 for config in %{vgaconfigs}; do
     build_bios %{_sourcedir}/config.vga-${config} \
                vgabios.bin vgabios-${config}.bin out/vgabios.bin
