diff --git a/spice-gtk.spec b/spice-gtk.spec
index 9bd79d1..a9e39a7 100644
--- a/spice-gtk.spec
+++ b/spice-gtk.spec
@@ -31,7 +31,9 @@ BuildRequires: gtk-doc
 BuildRequires: vala
 BuildRequires: usbutils
 BuildRequires: libsoup-devel >= 2.49.91
+%if 0%{?fedora}
 BuildRequires: libphodav-devel
+%endif
 BuildRequires: lz4-devel
 BuildRequires: gtk3-devel
 BuildRequires: json-glib-devel
@@ -105,6 +107,7 @@ Simple clients for interacting with SPICE servers.
 spicy is a client to a SPICE desktop server.
 spicy-screenshot is a tool to capture screen-shots of a SPICE desktop.
 
+%global debug_package %{nil}
 
 %prep
 gpgv2 --quiet --keyring %{SOURCE2} %{SOURCE1} %{SOURCE0}
@@ -120,6 +123,9 @@ gpgv2 --quiet --keyring %{SOURCE2} %{SOURCE1} %{SOURCE0}
 %ifarch s390x # https://gitlab.freedesktop.org/spice/spice-gtk/issues/120
   -Dusbredir=disabled \
 %endif
+%if 0%{?rhel}
+  -Dwebdav=disabled \
+%endif
 %if 0%{?flatpak}
   -Dpolkit=disabled
 %else
