diff --git a/spice-gtk.spec b/spice-gtk.spec
index 5c63fb6..e14863f 100644
--- a/spice-gtk.spec
+++ b/spice-gtk.spec
@@ -39,8 +39,10 @@ BuildRequires: polkit-devel
 BuildRequires: gtk-doc
 BuildRequires: vala
 BuildRequires: usbutils
+%if 0%{?fedora}
 BuildRequires: libsoup3-devel
 BuildRequires: libphodav-devel >= 3.0
+%endif
 BuildRequires: lz4-devel
 BuildRequires: gtk3-devel
 BuildRequires: json-glib-devel
@@ -114,6 +116,7 @@ Simple clients for interacting with SPICE servers.
 spicy is a client to a SPICE desktop server.
 spicy-screenshot is a tool to capture screen-shots of a SPICE desktop.
 
+%global debug_package %{nil}
 
 %prep
 #gpgv2 --quiet --keyring %{SOURCE2} %{SOURCE1} %{SOURCE0}
@@ -129,6 +132,9 @@ spicy-screenshot is a tool to capture screen-shots of a SPICE desktop.
 %ifarch s390x # https://gitlab.freedesktop.org/spice/spice-gtk/issues/120
   -Dusbredir=disabled \
 %endif
+%if 0%{?rhel}
+  -Dwebdav=disabled \
+%endif
 %if 0%{?flatpak}
   -Dpolkit=disabled
 %else
