diff --git a/virt-manager.spec b/virt-manager.spec
index 36b9449..507da5c 100644
--- a/virt-manager.spec
+++ b/virt-manager.spec
@@ -27,6 +27,7 @@ Requires: python3-gobject >= 3.31.3
 Requires: gtk3 >= 3.22.0
 Requires: libvirt-glib >= 0.0.9
 Requires: gtk-vnc2
+Requires: spice-gtk3
 
 # virt-manager is one of those apps that people will often install onto
 # a headless machine for use over SSH. This means the virt-manager dep
@@ -124,7 +125,7 @@ git commit -q -a --allow-empty --author 'rpm-build <rpm-build>' -m symlinks
 
 %build
 %if 0%{?rhel}
-%global _default_graphics -Ddefault-graphics=vnc
+%global _default_graphics -Ddefault-graphics=spice
 %endif
 
 %meson \
